src/lib/flash-helper.test.js

```diff
@@ -217,6 +267,24 @@ describe('flash-helper', () => {
 			expect(steps).to.deep.equal(expected);
 		});
 
+		it('returns a list of flash steps for gen3 nrf52 based platform', async () => {
+			const steps = await createFlashSteps({
+				modules,
+				platformId: 13,
+				isInDfuMode: false,
+			});
+
+			const expected = [
+				preBootloaderStep,
+				bootloaderStep,
+				systemPart1Step,
+				systemPart2Step,
+				userPart1Step,
+				userPartInvalidationStep
+			];
```

Comment:
The expected order is
```suggestion
			const expected = [
				preBootloaderStep,
				bootloaderStep,
				systemPart1Step,
				systemPart2Step,
				userPartInvalidationStep,
				userPart1Step
			];
```